presto-native-execution/presto_cpp/main/types/PrestoToVeloxConnector.cpp

```diff
@@ -0,0 +1,1388 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "PrestoToVeloxConnector.h"
+#include <velox/type/fbhive/HiveTypeParser.h>
+#include "velox/connectors/hive/HiveConnector.h"
+#include "velox/connectors/hive/HiveConnectorSplit.h"
+#include "velox/connectors/hive/HiveDataSink.h"
+#include "velox/connectors/hive/TableHandle.h"
+#include "velox/connectors/hive/iceberg/IcebergDeleteFile.h"
+#include "velox/connectors/hive/iceberg/IcebergSplit.h"
+#include "velox/connectors/tpch/TpchConnector.h"
+#include "velox/connectors/tpch/TpchConnectorSplit.h"
+
+namespace facebook::presto {
+
+namespace {
+std::vector<std::unique_ptr<const PrestoToVeloxConnector>>& connectors() {
+  static std::vector<std::unique_ptr<const PrestoToVeloxConnector>> connectors;
+  return connectors;
+}
```

Comment:
`getPrestoToVeloxConnectors` is declared to return a `const` vector to avoid potential modifications (to ensure it is only modified via `registerPrestoToVeloxConnector`)